src/repositories/consent.ts

```diff
@@ -19,27 +19,47 @@
  - Name Surname <name.surname@mojaloop.io>
 
  * Google
- - Steven Wijaya <stevenwjy@google.com>
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
  --------------
  ******/
 
-/* eslint-disable @typescript-eslint/no-explicit-any */
+/* istanbul ignore file */
+// TODO: BDD Testing will covered in separate ticket #1702
 
 import firebase from '~/lib/firebase'
 import { Consent } from '~/models/consent'
 import { logger } from '~/shared/logger'
 
 export interface IConsentRepository {
+  /**
+   * Updates a consent document based on a unique identifier.
+   *
+   * @param id    Id for the consent document that needs to be updated.
+   * @param data  Document fields that are about to be updated.
+   */
+  updateConsentById(id: string, data: Record<string, unknown>): Promise<void>
```

Comment:
shouldn't we use the Consent interface as data type here? Record<string, unknown> is very generic

Reply:
Hi - this has been done intentionally as we are not passing in the entire consent object everytime - we pass in just the key/values that need to be updated

Reply:
if we pass fields that aren't present in the Consent interface - Will they be blindly stored? What happens if by mistake we will pass property of a different type instead?

Reply:
Any external objects should be getting validated by the JOI validation + for firestore handlers - that's coming from the db itself - only other place is when we process internally and update
but still I agree we need to be more defensive here - maybe I can add in a check somewhere 
@stevenwjy was wondering if you have any thoughts

Reply:
The reason for not using the `Consent` interface was because the interface has some non-optional variables, which may not be suitable for performing the update since we only want to update some specific properties which may or may not be optional... But I agree that we should be more defensive here

Reply:
Writing an issue on this 
We should ensure firebase security rules are in place and do as much as we can with those, else log issues - and can run an internal validator on it

Reply:
https://github.com/mojaloop/pisp-demo-server/issues/new